pkg/odo/cli/component/create.go

```diff
@@ -235,6 +235,13 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 	if err != nil {
 		return errors.Wrap(err, "failed intiating local config")
 	}
+
+	// check to see if config file exists or not, if it does that
+	// means we shouldn't allow the user to override the current component
+	if co.localConfigInfo.ConfigFileExists() {
+		return errors.New("this directory already contains a component")
```

Comment:
This check is in the `Complete` phase because of https://github.com/openshift/odo/pull/1526#issuecomment-481162707